source/definitions.rst

```diff
@@ -32,13 +32,11 @@ Definitions
    Component
        A component is an :term:`object` or :term:`NTS object`.
 
-       A component is identified using component id.
+       A component is identified using a component id.
 
    Component id
-       Identity of a :term:`component`
-
-       The format used for the STA’s sites is specified in the STA
-       publication TDOK 2012:1171, e.g. AA+BBCDD=EEEFFGGG.
+       Identity of a :term:`component`.
+        A component id is a string, e.g. KK+AG0503=001DL001 or DL1.
```

Comment:
Perhaps we should explicit state that the last part, e.g. "DL001" or "DL1" should at least exist. I wonder if we by accepting any random format causes issues for existing implementations? They might assume that the last part with the consecutive numbering (e.g. 001, 002) always exist.

Reply:
Perhaps, but it would make it ackward to use more descriptive names lika bus_prio for a detector og car_right for a signal group?

Reply:
Yes. Perhaps. In principle I agree with you that any random format should be good enough. But It also means that the manufacturers needs to internally map each detector and signal group to their RSMP counterpart - something we've done implicitly by using consecutive numbering. From a protocol point of view this is OK, but I wonder what it means for the manufacturers? Is there a need for extra configuration in the equipment?

Reply:
I think all we have to do is mandate that each component have an id, and also an integer index. Both must be unique on the device, for that type of component. E.g. all signal groups msut have different ids and indexes.

Existing devices that use component ids that end with an integer part, like KK+AG0503=001DL001, can use that last part (0001) as the index, e.g TLCs in the Nordics.

Other devices that allow arbitrary strings as component ids, e.g. "dl/bus" will have to maintain index using some other mechanism. We should should recommend against alphabetical ordering for indexing, as indexes can jump around when you add/remove/rename components.